packages/contracts-bedrock/test/libraries/EOA.t.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Forge
+import { Test } from "forge-std/Test.sol";
+
+// Libraries
+import { EOA } from "src/libraries/EOA.sol";
+
+/// @title EOA_Helper
+/// @notice A helper contract to test the EOA library.
+contract EOA_Helper {
+    /// @notice Returns true if the sender is an EOA.
+    /// @return isEOA_ True if the sender is an EOA.
+    function isSenderEOA() external view returns (bool isEOA_) {
+        return EOA.isSenderEOA();
+    }
+}
+
+/// @title EOA_isEOA_Test
+contract EOA_isEOA_Test is Test {
+    EOA_Helper helper;
+
+    /// @notice Sets up the test.
+    function setUp() public {
+        helper = new EOA_Helper();
+    }
+
+    /// @notice Tests that a standard EOA is detected as an EOA.
+    /// @param _privateKey The private key of the sender.
+    function testFuzz_isEOA_isStandardEOA_succeeds(uint256 _privateKey) external {
+        // Make sure that the private key is in the range of a valid secp256k1 private key.
+        _privateKey = bound(_privateKey, 1, SECP256K1_ORDER - 1);
+
+        // Make sure that the sender is a standard EOA with no code.
+        address sender = vm.addr(_privateKey);
+        vm.assume(sender.code.length == 0);
+
+        // Should be considered an EOA
+        vm.prank(sender, sender);
+        assertEq(helper.isSenderEOA(), true);
+    }
+
+    /// @notice Tests that a 7702 EOA is detected as an EOA.
+    /// @param _privateKey The private key of the sender.
+    /// @param _7702Target The target of the 7702 EOA.
+    function testFuzz_isEOA_is7702EOA_succeeds(uint256 _privateKey, address _7702Target) external {
+        // Make sure that the private key is in the range of a valid secp256k1 private key.
+        _privateKey = bound(_privateKey, 1, SECP256K1_ORDER - 1);
+
+        // Make sure that the sender is a 7702 EOA.
+        address sender = vm.addr(_privateKey);
+        vm.etch(sender, abi.encodePacked(hex"EF0100", _7702Target));
+
+        // Should be considered a 7702 EOA.
+        vm.prank(sender);
+        assertEq(helper.isSenderEOA(), true);
```

Comment:
we should also do this with a different origin to hit both branches of the 7702 case
```suggestion
        // Should be considered a 7702 EOA.
        vm.prank(sender, sender);
        assertEq(helper.isSenderEOA(), true);

        vm.prank(sender, someOtherTxOrigin);
        assertEq(helper.isSenderEOA(), true);
```